### PR TITLE
chore: resolve conflicts for `daan/api`

### PR DESCRIPTION
**Pull request by chungquantin**
Rebase and update the branch `daan/api` to the latest version of the `master` branch.

### CODE REVIEW

File: extension/src/functions.rs [hunk 1]
```diff
@@ -350,23 +354,28 @@ mod tests {
 			})
 		}
 
-		#[test]
+#[test]
 		fn dispatch_call_adjusts_weight() {
```
Comment (chungquantin) [inline extension/src/functions.rs:358]: > [!IMPORTANT]
> Behavior of weight charges in the pallet-contracts is updated which impacts directly to the `dispatch_call_adjust_weight_works` test. 
> - Reference: https://github.com/paritytech/polkadot-sdk/commit/b6231c79ca708d9c9280210f41ca38fd816c8ad9
> - No change to pre-dispatch weight charged

File: pop-api/integration-tests/src/fungibles/mod.rs [hunk 2]
```diff
@@ -569,7 +569,7 @@ fn burn_works() {
 		start_destroy_asset(addr.clone(), asset);
 		assert_eq!(
 			burn(addr.clone(), asset, BOB, amount),
-			Err(Module { index: 52, error: [16, 0] })
+			Err(Module { index: 52, error: [17, 0] })
```
Comment (chungquantin) [inline pop-api/integration-tests/src/fungibles/mod.rs:572]: > [!Note]
> Index of `burn()` method is updated from `16` to `17`

**Review by peterwht — APPROVED** [review]
Amazing!! So close to getting this fully merged

File: extension/src/mock.rs [hunk 3]
```diff
@@ -80,31 +82,39 @@ impl pallet_balances::Config for Test {
...
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeHoldReason = RuntimeHoldReason;
+	type Schedule = MySchedule;
+	type Time = Timestamp;
+	type UnsafeUnstableInterface = ();
+	// UnstableInterface;
+	type UploadOrigin = EnsureSigned<Self::AccountId>;
+	type WeightInfo = ();
+	type WeightPrice = ();
+	// Self;
```
Comment (evilrobot-01) [inline extension/src/mock.rs:117]:
```suggestion

```

**Review by evilrobot-01 — APPROVED** [review]
Seems fine, a lot of noise due to the large difference between main and the target branch. Will be easier to navigate once merged so think we should proceed as is and be more thorough with the final review.

**Pull request merged**